TCMSApp/src/server/model/mschemas/defect.js

```diff
@@ -1,22 +1,24 @@
 //import the necessary modules
 var mongoose = require('mongoose');
+var uniqueValidator = require('mongoose-unique-validator');
 var Schema = mongoose.Schema;
 
 //create  the defect model
 //define schema
 var defectSchema = new Schema({
+        randomId: Number,
         name: {type: String, required: true, unique: true},
-        reporter: {type: mongoose.Schema.Types.ObjectId, required: true},
-        assignedTo: {type: mongoose.Schema.Types.ObjectId},
-        date: Date,
+        whoFind: String,
+        assignedTo: String,
+        dateOfDefectCreation: Date,
         priority: {
             type: String,
             required: true,
             enum: ['Critical', 'Low', 'High', 'Normal']
         },
         description: {type: String, required: true},
-        stepsToReproduce: [String],
-        run: {type: mongoose.Schema.Types.ObjectId}
-    });
-var defect = mongoose.model('Defect', defectSchema) ;
+        stepsToReproduce:[String],
+        testRunId:Number});
```

Comment:
We have already type: mongoose.Schema.Types.ObjectId for run. Maybe it's the same as testRunId